apps/mobile/src/screens/NftDetailScreen/NftDetailAsset/NftDetailAssetCacheSwapper.tsx

```diff
@@ -75,28 +72,48 @@ export function NftDetailAssetCacheSwapper({ children, style, cachedPreviewAsset
         ]}
       >
         {fullyRemoved ? null : (
-          <NftPreviewAsset
+          <RawNftPreviewAsset
             onLoad={assetSizer.handleLoad}
-            tokenUrl={cachedPreviewAssetUrl}
+            tokenUrl={cachedPreviewAssetUrl ?? ''}
             resizeMode={ResizeMode.CONTAIN}
           />
         )}
       </View>
+    );
+  }, [
+    assetSizer.finalAssetDimensions,
+    assetSizer.handleLoad,
+    assetSizer.handleViewLayout,
+    cachedPreviewAssetUrl,
+    detailLayoutHeight,
+    fullyRemoved,
+    maxHeight,
+  ]);
+
+  if (!cachedPreviewAssetUrl) {
+    return <>{children}</>;
+  }
+
+  return (
+    <View style={[style, { position: 'relative' }]}>
+      <ReportingErrorBoundary fallback={preview}>
+        {preview}
```

Comment:
very excited for this one. notice the `fallback={preview}` and the `{preview}` rendered underneath.

the way this component works is:
1. immediately display preview image URL while main asset loads
2. attempt to render main asset
3. if fails, fall back to the preview image

this will handle unhandled cases such as figure31's glb file type, which by default breaks on mobile currently

| Before | After |
|--------|--------|
| <img width="325" alt="image" src="https://github.com/gallery-so/gallery/assets/12162433/addf7f5a-5230-42df-9af9-04a36cf2c57e"> | ![image](https://github.com/gallery-so/gallery/assets/12162433/716fb3c9-8f51-4d88-a661-0a3bf2f7aac9) |

Reply:
massiveee